scripts/validate-artifacts.ps1

```diff
@@ -34,138 +34,155 @@ Write-Output "Validating artifacts"
-    }
-    if ($virtualMachine.properties.storageProfile.imageReference.offer.Contains('ubuntu-server') -and $virtualMachine.properties.storageProfile.imageReference.sku.Contains('22_04')) { 
-        Write-Output "`u{2705} Checked Virtual Machine OS image offer - OK"
+if ($nic.properties.ipConfigurations.Count -eq 1) { 
+    if ($nic.properties.ipConfigurations.properties.publicIPAddress -and $nic.properties.ipConfigurations.properties.publicIPAddress.id) { 
+        Write-Output "`u{2705} Checked if Public IP assigned to the VM - OK"
     } else { 
         Write-Output `u{1F914}
-        throw "Virtual Machine uses wrong OS image. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again." 
+        throw "Unable to verify Public IP configuratio for the VM. Please make sure that IP configuration of the VM network interface has public IP address configured and try again."
     }
+} else {
+    Write-Output `u{1F914}
+    throw "Unable to verify IP configuration of the Network Interface. Please make sure that you have 1 IP configuration of the VM network interface and try again."
+}
 
-    if ($virtualMachine.properties.hardwareProfile.vmSize -eq "Standard_B1s") { 
-        Write-Output "`u{2705} Checked Virtual Machine size - OK"
-    } else { 
+
+$nsg = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/networkSecurityGroups")
+if ($nsg) {
+    if ($nsg.name.Count -eq 1) { 
+        Write-Output "`u{2705} Checked if the Network Security Group resource exists - OK"
+    }  else { 
         Write-Output `u{1F914}
-        throw "Virtual Machine size is not set to B1s. Please make sure that your script creates a VM with size B1s and try again."
+        throw "More than one Network Security Group resource was found in the VM resource group. Please delete all un-used Network Security Group resources and try again."
     }
+} else {
+    Write-Output `u{1F914}
+    throw "Unable to find Network Security Group resouce. Please re-deploy the VM and try again."
 }
 
-if ($virtualMachines[0].zones -ne $virtualMachines[1].zones) { 
-    Write-Output "`u{2705} Checked Virtual Machines deployed across different availability zones - OK"
+$sshNsgRule = ( $nsg.properties.securityRules | Where-Object { ($_.properties.destinationPortRange -eq '22') -and ($_.properties.access -eq 'Allow')} ) 
+if ($sshNsgRule)  {
+    Write-Output "`u{2705} Checked if NSG has SSH network security rule configured - OK"
 } else { 
     Write-Output `u{1F914}
-    throw "Virtual Machines are deployed to the same availability zone. Please make sure that you are assigning distinct availabilty zones during the VM creation with parameter '-Zone' and try again. "
+    throw "Unable to fing network security group rule which allows SSH connection. Please check if you configured VM Network Security Group to allow connections on 22 TCP port and try again."
```

Comment:
There is a typo in the error message: 'fing' should be 'find'. Correcting this will make the error message clearer.